crates/e2e/src/setup/mod.rs

```diff
@@ -82,69 +94,52 @@ where
     F: FnOnce(Web3) -> Fut,
     Fut: Future<Output = ()>,
 {
-    let filters = DEFAULT_FILTERS
-        .map(|s| s.to_string())
-        .into_iter()
-        .collect::<Vec<_>>();
-
-    run(f, filters, None, None).await
+    run(f, None::<Vec<String>>, None, None).await
 }
 
-pub async fn run_test_with_extra_filters<F, Fut>(f: F, extra_filters: Vec<String>)
-where
+pub async fn run_test_with_extra_filters<F, Fut, T>(
+    f: F,
+    extra_filters: impl IntoIterator<Item = T>,
+) where
     F: FnOnce(Web3) -> Fut,
     Fut: Future<Output = ()>,
+    T: AsRef<str>,
 {
-    let filters = DEFAULT_FILTERS
-        .map(|s| s.to_string())
-        .into_iter()
-        .chain(extra_filters.into_iter())
-        .collect::<Vec<_>>();
-
-    run(f, filters, None, None).await
+    run(f, Some(extra_filters), None, None).await
 }
 
 pub async fn run_forked_test<F, Fut>(f: F, solver_address: H160, fork_url: String)
 where
     F: FnOnce(Web3) -> Fut,
     Fut: Future<Output = ()>,
 {
-    let filters = DEFAULT_FILTERS
-        .map(|s| s.to_string())
-        .into_iter()
-        .collect::<Vec<_>>();
-
-    run(f, filters, Some(solver_address), Some(fork_url)).await
+    run(f, None::<Vec<String>>, Some(solver_address), Some(fork_url)).await
 }
 
-pub async fn run_forked_test_with_extra_filters<F, Fut>(
+pub async fn run_forked_test_with_extra_filters<F, Fut, T>(
     f: F,
     solver_address: H160,
     fork_url: String,
-    extra_filters: Vec<String>,
+    extra_filters: impl IntoIterator<Item = T>,
 ) where
     F: FnOnce(Web3) -> Fut,
     Fut: Future<Output = ()>,
+    T: AsRef<str>,
 {
-    let filters = DEFAULT_FILTERS
-        .map(|s| s.to_string())
-        .into_iter()
-        .chain(extra_filters.into_iter())
-        .collect::<Vec<_>>();
-
-    run(f, filters, Some(solver_address), Some(fork_url)).await
+    run(f, Some(extra_filters), Some(solver_address), Some(fork_url)).await
 }
 
-async fn run<F, Fut>(
+async fn run<F, Fut, T>(
     f: F,
-    filters: Vec<String>,
+    filters: Option<impl IntoIterator<Item = T>>,
```

Comment:
This does not need the `Option`. If you don't want to pass additional filter you can pass `std::iter::empty()` or probably just `[]`.